ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/transaction/UtxoFilteredTransaction.kt

```diff
@@ -0,0 +1,69 @@
+package net.corda.v5.ledger.utxo.transaction
+
+import net.corda.v5.base.annotations.DoNotImplement
+import net.corda.v5.crypto.SecureHash
+import net.corda.v5.ledger.common.Party
+import net.corda.v5.ledger.common.transaction.TransactionMetadata
+import net.corda.v5.ledger.common.transaction.TransactionVerificationException
+import net.corda.v5.ledger.utxo.Command
+import net.corda.v5.ledger.utxo.StateAndRef
+import net.corda.v5.ledger.utxo.StateRef
+import net.corda.v5.ledger.utxo.TimeWindow
+import java.security.PublicKey
+
+@DoNotImplement
+interface UtxoFilteredTransaction {
+    /**
+     * @property id The ID of the transaction.
+     */
+    val id: SecureHash
+
+    /**
+     * @property metadata the metadata for this transaction
+     */
+    val metadata: TransactionMetadata
+
+    /**
+     * @param timeWindow The validity time window for finalizing/notarising this transaction or null if filtered
+     */
+    val timeWindow: TimeWindow?
+
+    /**
+     * @param notary The notary party for this transaction or null if filtered
+     */
+    val notary: Party?
```

Comment:
Why are these null, as opposed to being a piece of filtered data using the "removed" interface?

This is particularly concerning for time windows, because we rely on the upper bound being part of the merkle tree to prevent state reissuance after being archived.

Reply:
They are nullable because technically, they are part of the same component group, but on the interfaces for transactions, they are separate, and they are never lists, always just one entry. 
So it's mostly a convenience thing that they are just nullable vals rather than being put through the whole removed interface shenanigans (and a size proof makes inherentely no sense).
They can technically be filtered, but that would obviously be wrong if they do get filtered out for a non-validating notary payload.

Reply:
More generally though, I guess its possible the transaction tear off use-case might want to filter them?

I see now though what you mean about its only list based data that is using the filtered data objects. In that case, either the naming or the abstraction doesn't quite feel right to me, if filtered data is specifically filtered _list_ data. It doesn't seem consistent from an API perspective to have two ways of checking for absence depending on whether the data of interest is a list or not. Maybe what we're missing is a fourth filtered data type - we need the current 3 - absent, size (only makes sense to lists), _list_ data audit, and a new _single_ data audit. Or capture audit single items as a list of size 1.

Reply:
Hmm, I am not quite sure I follow here - clearly the UtxoFilteredTransaction has some more specifications, so we know a priori that the SignedTransaction has one notary and one time window - so in the filtered transaction, they are either present or filtered out (null).

Reply:
Yeah, I'm not saying that what's there doesn't work. It's just putting my API user hat on, it feels a bit inconsistent - that _how_ I check for whether something is filtered or not depends on whether the item is a list or not. And the other assumption which you've touched on, is that the null approach only works for items that are mandatory. That may be a safe assumption for notary and time window when constructing a filtered transaction for the notary to consume, but not 100% sure if it would hold for transaction tear offs between peers. Also, if we extend this interface in future with (non-list) optional attributes, then we would have no ability to distinguish between "filtered" and "supplied but absent".

Personally I would ensure that there is an explicit "filtered" type for everything, rather than relying on null for singular arguments, but it's your call.